docs/en/04_Changelogs/4.1.0.md

```diff
@@ -0,0 +1,188 @@
+- index.php
+- install.php
+```
+
+After migration the folder structure would look like:
+
+```
+- mysite/
+  - code/
+    - Page.php
+    - PageController.php
+- public/
+  - assets/
+    - .protected/
+  - css/
+    - somestyle.css
+  - resources/ (auto-generated by vendor-plugin `composer vendor-expose` command)
+    - themes/
+      - mytheme/
+        - css/
+          - style.css
+    - vendor
+      - silverstripe/
+        - blog/
+          - css/
+            - style.css
+  - .htaccess
+  - favicon.ico
+  - index.php
+  - install.php
+- themes/
+  - mytheme/
+    - css/ (exposed in composer.json)
+      - style.css
+    - templates/
+      - BlogPage.ss
+- vendor/
+  - silverstripe/
+    - blog/
+      - css/ (exposed in composer.json)
+        - style.css
+```
+ 
+### Use new `$public` theme set
+ 
+In addition there is a new helper pseudo-theme that you can configure to expose files in the `public/`
+folder to the themed css / javascript file lookup. For instance, this is how you can prioritise those
+files:
+
+```yaml
+---
+Name: mytheme
+---
+SilverStripe\View\SSViewer:
+  themes:
+    - '$public'
+    - 'simple'
+    - '$default'
+```
+
+This would allow `<% require themedCSS('style.css') %>` to find a file comitted to `public/css/style.css`.
+
+Note that `Requirements` calls will look in both the `public` folder (first) and then the base path when
+resolving css or javascript files. Any files that aren't in the public folder must be exposed using
+the composer.json "expose" mechanism described below.
+
+### Expose root project files
+
+If you have files comitted to source control outside of the `public` folder, but you need them to be available
+to the web server, you can also use the composer.json `expose` directive to symlink / copy these to `public/resources/`.
+
+**composer.json** (in project root)
+
+```json
+{
+  "extra": {
+    "expose": [
+      "mysite/client"
+    ]
+  }
+}
+```
+
+Then run the composer helper `composer vendor-expose` in your project root. This will symlink (or copy)
+the `mysite/client` directory to `public/resources/mysite/client`.
+
+If you are using third party modules which may not have explic `expose` directives,
+you can also expose those assets manually by declaring the full path to the directory to expose.
+This works the same for `silverstripe-module` and `silverstripe-vendormodule` types.
+
+```json
+{
+  "extra": {
+    "expose": [
+      "vendor/somevendor/somemodule/css",
+      "anothermodule/css"
+    ]
+  }
+}
+```
+
+### Path manipulation helpers
+
+The following filesystem helpers have been added in order to better support working with cross-platform
+path manipulation:
+
+ * `SilverStripe\Core\Path::join()` which will join one or more relative or absolute paths
+ * `SilverStripe\Core\Path::normalise()` which will normalise and trim directory separators in a relative or absolute path
+ * `SilverStripe\Core\Convert::slashes()` to convert directory separators to either `/` or `\`
```

Comment:
These descriptions are a bit obtuse. Example inputs and outputs for each would really help.